python-avd/tests/pyavd/j2filters/test_list_compress.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import pytest
+from pyavd.j2filters.list_compress import list_compress
+
+
+class TestListCompressFilter:
+    @pytest.mark.parametrize(
+        "invalid_list_to_compress, error_msg",
+        [
+            ("1-3", "Value must be of type list, got <class 'str'>"),
+            ({"key": "value"}, "Value must be of type list, got <class 'dict'>"),
+            (33, "Value must be of type list, got <class 'int'>"),
+            (["a", "b", "c"], "All elements of the list ['a', 'b', 'c'] must be integers"),
```

Comment:
can we include one more test with different data types like [1, "a"]

Reply:
added!